udmis/src/main/java/com/google/bos/udmi/monitoring/LogTail.java

```diff
@@ -0,0 +1,253 @@
+package com.google.bos.udmi.monitoring;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.audit.AuditLog;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.LogEntryServerStream;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Logging.TailOption;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Payload;
+import com.google.cloud.logging.Severity;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.time.Instant;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import udmi.schema.Monitoring;
+import udmi.schema.MonitoringMetric;
+
+public class LogTail {
+
+  private String projectId;
+  private LogTimeSeries logsTimeSeries;
+  private String LOG_FILTER =
+      "(resource.labels.function_name=\"udmi_target\") OR " +
+          "(resource.labels.function_name=\"udmi_state\") OR " +
+          "(resource.labels.function_name=\"udmi_config\") OR " +
+          "(severity=ERROR AND protoPayload.serviceName=\"cloudiot.googleapis.com\")";
+
+  public LogTail(String projectId) {
+    this.projectId = projectId;
+    this.logsTimeSeries = new LogTimeSeries();
+  }
+
+  public static void main(String[] args) {
+    String projectId = "essential-keep-197822";
+    LogTail lt = new LogTail(projectId);
+    lt.tailLogs();
+  }
+
+  private LogEntryServerStream getCloudLogStream(String filter) {
+    LoggingOptions options = LoggingOptions.getDefaultInstance();
+    Logging logging = options.getService();
+    LogEntryServerStream stream;
+    if (filter == null) {
+      stream = logging.tailLogEntries();
+    } else {
+      stream = logging.tailLogEntries(TailOption.filter(filter));
+    }
+    return stream;
+  }
+
+  /*
+   * Top level processor of incoming GCP log entries.
+   */
+  private void processLogEntry(LogEntry log) {
```

Comment:
unit test?

Reply:
Will start writing them. Sending the class refactoring in parallel.